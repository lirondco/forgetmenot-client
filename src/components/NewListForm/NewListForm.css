.insertList {
    padding: 5%;
}

.insertList h3 {
    text-align: center;
    margin: 5%;
    font-family: 'Patrick Hand SC', cursive;
    font-size: 21pt;
}

.insertList form {
    display: flex;
    flex-direction: column;
}

.insertList form select {
    margin-bottom: 5%;
    width: fit-content;
}