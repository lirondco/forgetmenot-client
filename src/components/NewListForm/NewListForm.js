import React, { Component } from 'react'
import ListContext from '../../contexts/ListContext'
import IdeasApiService from '../../services/ideas-api-service'
import { Input, Button, Required } from '../Utils/Utils'
import './NewListForm.css'

export default class NewListForm extends Component {
  static defaultProps = {
    onPostListSuccess: () => {}
}
  static contextType = ListContext 
  
  handleSubmitList = ev => {
    ev.preventDefault()
    const { name } = ev.target

    IdeasApiService.postList(name.value)
      .then(this.context.addList)
      .then(() => {
        name.value = ''
        this.props.onPostListSuccess()
      })
      .catch(this.context.setError)
  }

  render() {
    return (
    <div className='insertList'>
    <h3>You're entering a new list for your ideas</h3>
      <form
        className='listForm'
        onSubmit={this.handleSubmitList}
      >
        <div className='listName'>
            <label htmlFor='name'>New list's name:</label>
            <Required />
            <Input
                required
                type='text'
                maxLength='30'
                name='name'
                id='name'
                />
        </div>
        <Button type='submit'>
          Add List
        </Button>
      </form>
    </div>
    )
  }
}
