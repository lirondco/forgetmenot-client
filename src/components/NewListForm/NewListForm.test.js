import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NewListForm from './NewListForm'

describe(`NewListForm component`, () => {
  it('renders the complete form', () => {
    const wrapper = shallow(<NewListForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})