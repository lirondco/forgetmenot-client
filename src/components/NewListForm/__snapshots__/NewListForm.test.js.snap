// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewListForm component renders the complete form 1`] = `
<div
  className="insertList"
>
  <h3>
    You're entering a new list for your ideas
  </h3>
  <form
    className="listForm"
    onSubmit={[Function]}
  >
    <div
      className="listName"
    >
      <label
        htmlFor="name"
      >
        New list's name:
      </label>
      <Required />
      <Input
        id="name"
        maxLength="30"
        name="name"
        required={true}
        type="text"
      />
    </div>
    <Button
      type="submit"
    >
      Add List
    </Button>
  </form>
</div>
`;
